### ISSUE TITLE
Add support for 'default' fg/bg colors in ScreenBuffer

### ISSUE THREAD

**Issue by samizdatco** [cronvel/terminal-kit#30]
In experimenting with the ScreenBuffer object, it seems that even if I don't set an initial 'fill' color, when the draw() method is called and it updates the screen, it defaults to drawing a black background for every character. 

For instance, I would have expected these code snippets to result in comparable output:

```js
// directly draw to terminal
var termkit = require( 'terminal-kit' ),
    term = termkit.terminal;

term.clear()
term.moveTo(14,4, 'hello world')
term.moveTo(40, 9)
```
--------------------
```js
// indirectly draw via ScreenBuffer
var termkit = require( 'terminal-kit' ),
    term = termkit.terminal,
    ScreenBuffer = termkit.ScreenBuffer;

var screen = ScreenBuffer.create({
    dst:term,
    x:1, y:1,
    width:40,
    height:9
})
term.clear()
screen.put({x:14,y:4}, 'hello world')
screen.draw()
```

But the results are quite different looking:
![termkit-screenbuffer](https://cloud.githubusercontent.com/assets/469523/26705486/7dda48e4-4704-11e7-990f-ff0416aa0b17.png)

Is there any way that `default` or `transparent` could be added as a `bgColor` value in ScreenBuffer's `attrs` arg so that text could be displayed using the terminal's standard color scheme rather than overriding it? Or does this functionality already exist and I'm just misunderstanding how to use it?

**Comment by cronvel** [comment]
@samizdatco In fact, there **IS** a call to ScreenBuffer#fill() at instanciation, unless the 'noFill' option is passed to [ScreenBuffer.create()](https://github.com/cronvel/terminal-kit/blob/master/doc/screenbuffer.md#ref.ScreenBuffer.create). Without that initial .fill(), the internal Buffer instance is not filled, and is left in the state of it was after [Buffer.allocUnsafe()](https://nodejs.org/dist/latest-v6.x/docs/api/buffer.html#buffer_class_method_buffer_allocunsafe_size): this option is usually not recommended except if you call .fill() immediately with your own option, or if you draw immediately the whole surface (e.g. with other screenBuffers, with ScreenBuffer#put(), etc). In case you don't, your screenBuffer is simply filled with garbage.

Also, the transparency option does not works when drawing directly to a terminal. You know, you just can't get informations about the content of a terminal, so it's impossible to implement transparency. And it's impossible to guess the actual background color for a given cell, neither the actual character displayed.

Transparency only works when drawing a screenBuffer into another screenBuffer.

Basically, what you want to achieve should be done with a call to [ScreenBuffer#fill()](https://github.com/cronvel/terminal-kit/blob/master/doc/screenbuffer.md#ref.ScreenBuffer.fill) with the correct color, if you can guess it.

Using the screenBuffer is like opening a window in your window manager, the new window doesn't know what is below, and just start by clearing itself.

**Comment by samizdatco** [comment]
That does make sense. In my case though I'm not trying to *avoid* filling the screenbuffer, it's more that I'm not sure how to set the fg & bg colors to a pair of specific values. What I'd like to do is use the 'default text' and 'default background' colors for the terminal. 

From the Wikipedia page on [ANSI Escape Codes](https://en.wikipedia.org/wiki/ANSI_escape_code#graphics), it's this pair of values in particular that I'm after:
![ansi-sgr-params](https://cloud.githubusercontent.com/assets/469523/26738448/251c434a-479c-11e7-9845-0305bb625234.png)

Is there currently a way to tell ScreenBuffer to use these colors (rather than the ones defined with codes 30-38 and 40-48)?

**Comment by cronvel** [comment]
I'm afraid, but if that default color is not one of the 256 color palette, it wont be possible.
The ScreenBuffer uses one byte to code the BG color index and one byte to store the FG color index, there is no room for a 257th index. 

Usually, the default color is either black or white or a shade of black/white, inside the 8 ANSI color palette or the 16 aixterm color palette. But some terminals allow any true color as the default color even if it is not part of the 256 color palette.

I understand that it can be frustrating, I need to think about a solution...

**Comment by samizdatco** [comment]
I took a swing at adding the feature [here](#31). In my limited testing (and with my limited understanding of terminal-kit's internals) it *seems* like it's working. 

It wouldn't surprise me at all if there are bugs in the way transparency and blending are handled though since I don't really have a handle on what's happening in those methods. That said, the `spaceship.js` demo seems to be working identically, so to the extent that it exercises all of ScreenBuffer's features, that's a good sign...

**Comment by cronvel** [comment]
Thanks for your contribution, but unfortunately I was already working on that.
I'm trying to avoid a SemVer Major change and aim for a backward compatible change.

**Comment by cronvel** [comment]
@samizdatco it's done. It landed on v1.10.0.

You can do it using the new **attr** options `{defaultColor: true, bgDefaultColor: true }`.
This is also the default options for .fill() and at ScreenBuffer/ScreenBufferHD creation.
Tell me if that's working for you.

**Comment by samizdatco** [comment]
The code is looking good! I've found one place where the default colors are 'leaking' outside of their defined areas though. See #32

**Comment by cronvel** [comment]
Thanks, it's merged. I fixed ScreenBufferHD too.
